app/jobs/v3/services/update_broker_job.rb

```diff
@@ -65,7 +68,7 @@ def perform
           @warnings
         rescue => e
           begin
-            broker.update(state: previous_broker_state)
+            initial_broker.update(state: previous_broker_state)
           rescue
             raise CloudController::Errors::V3::ApiError.new_from_details('ServiceBrokerGone') if broker.nil?
```

Comment:
Right, when we rewrite the second broker.update, we need to restructure it anyways, otherwise the unit test fails, which tests the case that the broker got deleted meanwhile.